lib/hanrahashiv.py

```diff
@@ -365,8 +365,8 @@ def ReplaceItemsInFormula(
     for i in range(len(formula)):
         new_formula[i].set_color(formula[i].color)
     if len(colors) > 0:
-        for i in items_indices:
-            new_formula[i].set_color(formula[i])
+        for i in range(len(items_indices)):
+            new_formula[items_indices[i]].set_color(colors[i])

```

Comment:
A similar thing has been #75 . Be cautious when merging. I think both are correct, but Gor's version seems better.